### ISSUE TITLE
prefer-destructuring should include or exclude certain variables so that projects can prefer to destructure some variables but not affect others

### ISSUE THREAD

**Issue by ajbogh** [eslint/eslint#12751]
**What rule do you want to change?**

prefer-destructuring should include or exclude certain variable names, such that a developer can choose which repetitive variables should be destructured by default while not affecting the entire project, or which variables never need to be destructured without adding eslint-ignore lines in many files.

**Does this change cause the rule to produce more or fewer warnings?**

This change prevents the rule from automatically updating every file that has non-destructured variables, allowing the user to include or exclude only those variable which they feel need to be modified.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option

**Please provide some example code that this change will affect:**

```js
{
  "rules": {
    "prefer-destructuring": ["error", {
      "VariableDeclarator": {
        "array": false,
        "object": true
      },
      "AssignmentExpression": {
        "array": false,
        "object": true
      }
    }, {
      "enforceForRenamedProperties": false,
      "include": ["this.props", "this.state"], // ignores everything else not defined in "include"
      "exclude": ["this.props.context"] // now ignored even though it was part of "include"
    }]
  }
}
```

**What does the rule currently do for this code?**

N/A

**What will the rule do after it's changed?**

It will allow the rule to be explicit as to which variables it cares about. By default the behavior is unchanged, but with the change the rule will allow greater granularity.

**Are you willing to submit a pull request to implement this change?**

Yes (given other commitments).

**Comment by kaicataldo** [comment]
Please fill out the sections that ask for code examples, as this will help us evaluate this proposal better. Thanks!
